### PR TITLE
chore: update rustc

### PR DESCRIPTION
**Pull request by ritchie46**

### CODE REVIEW

**Comment by messense** [comment]
FYI, cargo workspace metadata inheritance support was added in https://github.com/PyO3/maturin/pull/1131, I plan to release maturin v0.13.4 this week.

File: polars/polars-core/Cargo.toml [hunk 1]
```diff
@@ -145,52 +145,32 @@ docs-selection = [
...
 ndarray = { version = "0.15", optional = true, default_features = false }
-num = "^0.4"
+num.workspace = true
 once_cell = "1"
 polars-arrow = { version = "0.24.2", path = "../polars-arrow", features = ["compute"] }
 polars-utils = { version = "0.24.2", path = "../polars-utils" }
 rand = { version = "0.8", optional = true, features = ["small_rng", "std"] }
 rand_distr = { version = "0.4", optional = true }
-rayon = "1.5"
+rayon.worspace = true
```
Comment (messense) [inline polars/polars-core/Cargo.toml:167]: This should be `rayon.workspace = true`

Comment (ritchie46) [reply polars/polars-core/Cargo.toml:167]: Good catch! Fixed in a follow up PR.

**Comment by cjermain** [comment]
This is a great change! Glad to see the arrow2 dependency information consolidate.

**Comment by ritchie46** [comment]
Yes, really neat new feature.

**Pull request merged**